homeassistant/components/light/xiaomi_miio.py

```diff
@@ -725,3 +734,87 @@ def __init__(self, name, light, model, unique_id):
         except DeviceException as ex:
             self._available = False
             _LOGGER.error("Got exception while fetching the state: %s", ex)
+
+
+class XiaomiMiDeskLamp(XiaomiAbstractLight):
+    """Representation of a Xiaomi Mi LED Desk Lamp (MJTD01YL)"""
+
+    def __init__(self, name, light, model, unique_id):
+        """Initialize the light device."""
+        super().__init__(name, light, model, unique_id)
+
+        self._color_temp = None
+
+    @property
+    def color_temp(self):
+        """Return the color temperature."""
+        return self._color_temp
+
+    @property
+    def min_mireds(self):
+        """Return the coldest color_temp that this light supports."""
+        return int(1000000.0/4000.0)
+
+    @property
+    def max_mireds(self):
+        """Return the warmest color_temp that this light supports."""
+        return int(1000000.0/2700.0)
+
+    @property
+    def supported_features(self):
+        """Return the supported features."""
+        return SUPPORT_BRIGHTNESS | SUPPORT_COLOR_TEMP
+
+    async def async_turn_on(self, **kwargs):
+        """Turn the light on."""
+        if ATTR_COLOR_TEMP in kwargs:
+            color_temp = kwargs[ATTR_COLOR_TEMP]
+            color_temp_kelvin = int(1000000.0/float(color_temp))
+            _LOGGER.debug(
+                "Setting color temperature: "
+                "%s mireds (%sK)",
+                color_temp, color_temp_kelvin)
+
+            result = await self._try_command(
+                "Setting color temperature failed: %s K",
+                self._light.set_color_temp, color_temp_kelvin)
+
+            if result:
+                self._color_temp = color_temp
+
```

Comment:
Redundant newline.

Reply:
Thanks. I think it would make sense to have these things in pylint.